### PR TITLE
Add cluster spec & exclusions

### PR DESCRIPTION
**Pull request by HumairAK**
## This introduces a breaking change

- [ ] Yes
- [x] No

## Description

This PR adds exclusions for dh-prod cluster and cluster spec template. For OCP 4 we will do something similar, simply append the clusters list in prod vars and add/merge exclusions for resources that the project admin is not restricted to. There is also a sample rolebinding for argocd-manager SA that lives in another namespace.

### CODE REVIEW

File: examples/argocd-manager-rolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: authorization.openshift.io/v1
+kind: RoleBinding
+metadata:
+  name: argocd-manager-rolebinding
+roleRef:
+  name: admin
+subjects:
+- kind: ServiceAccount
+  name: argocd-manager
+  namespace: argocd-manager
```
Comment (anishasthana) [inline examples/argocd-manager-rolebinding.yaml:10]: Missing a newline at the end of the file

File: examples/argocd-manager-rolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: authorization.openshift.io/v1
+kind: RoleBinding
+metadata:
+  name: argocd-manager-rolebinding
```
Comment (anishasthana) [inline examples/argocd-manager-rolebinding.yaml:4]: Should you specify the namespace as well in the metadata?

File: objects/clusters.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    argocd.argoproj.io/secret-type: cluster
+  annotations:
+    managed-by: argocd.argoproj.io
+data:
+  config: |
+    {{ cluster.data.config | to_json | b64encode }}
+  name: {{ cluster.data.name | b64encode}}
+  namespaces: {{ cluster.data.namespaces | join(',') | b64encode }}
+  server: {{ cluster.data.server | b64encode }}
+type: Opaque
```
Comment (anishasthana) [inline objects/clusters.yaml:16]: Missing new line

File: examples/argocd-manager-rolebinding.yaml (hunk 2 above)
Comment (HumairAK) [reply examples/argocd-manager-rolebinding.yaml:4]: I wanted to keep it generic

Comment (anishasthana) [reply examples/argocd-manager-rolebinding.yaml:4]: I'd just make it something like `namespace: <application_namespace>`

**Comment by anishasthana** [comment]
Will instructions for how to use these files come in a follow-up PR?

**Comment by HumairAK** [comment]
> Will instructions for how to use these files come in a follow-up PR?

Yep

**Comment by HumairAK** [comment]
removed yamllint bear linting, too many conflicts with jinja2 text within yaml files

**Pull request merged**